src/AclRegs.cc

```diff
@@ -196,6 +197,7 @@ Acl::Init()
 
 #if USE_ADAPTATION
     RegisterMaker("adaptation_service", [](TypeName name)->ACL* { return new ACLStrategised<const char *>(new ACLAdaptationServiceData, new ACLAdaptationServiceStrategy, name); });
+    RegisterMaker("adaptation_rep_header", [](TypeName name)->ACL* { return new ACLStrategised<HttpHeader*>(new ACLHTTPHeaderData, new ACLAdaptationRepHeaderStrategy, name); });
```

Comment:
@rousskov , we have an important use-case of minimum footprint proxies. Adaptation is one feature those builds *do* want to remove. As such this ACL and others like it should remain within the USE_* wrappers.

Reply:
This particular `#ifdef` removes very little code and memory use while being inconsistent with the rest of the PR where the bulk of the same feature is added unconditionally.  We should not compile and document a feature unconditionally but then save a few bytes by disabling basic configuration support for that feature, making developer and admin jobs more difficult for virtually nothing.

I believe my change request offers the right logic for deciding whether to make this code line conditional. Please note that the change request does not say that the line has to be conditional or unconditional. The correct decision depends on the (unknown to me) answer to the question posed in the change request.

As for the the so called "minimum footprint" proxies support in general, somebody will need to define that concept and the corresponding optimization principles before we can meaningfully decide whether the Project should support that use case and what that support actually means. Otherwise, it is trivial to use that phrase to argue for `#ifdef`-ing 90% of Squid code!